neon_utils/signal_utils.py

```diff
@@ -0,0 +1,176 @@
+        LOG.warning("Initializing new messagebus connection")
+        init_signal_bus(MessageBusClient())
+    if _BUS.connected_event.wait(10):  # Wait up to 10 seconds for the bus service
+        response = _BUS.wait_for_response(Message("neon.signal_manager_active"))
+        LOG.debug(f"signal_manager_active={response is not None}")
+        return response is not None
+    LOG.error(f"Signal manager check gave up waiting for the MessageBus")
+    return False
+
+
+def _manager_create_signal(signal_name: str, *_, **__) -> bool:
+    """
+    Backwards-compatible method for creating a signal
+    :param signal_name: named signal to create
+    :return: True if signal exists
+    """
+    stat = _BUS.wait_for_response(Message("neon.create_signal",
+                                          {"signal_name": signal_name}),
+                                 f"neon.create_signal.{signal_name}", 10) or Message('')
+    return stat.data.get("is_set")
+
+
+def _manager_check_for_signal(signal_name: str, sec_lifetime: int = 0, *_, **__) -> bool:
+    """
+    Backwards-compatible method for checking for a signal
+    :param signal_name: name of signal to check
+    :param sec_lifetime: max age of signal in seconds before clearing it and returning False
+    :return: True if signal exists
+    """
+    stat = _BUS.wait_for_response(Message("neon.check_for_signal",
+                                          {"signal_name": signal_name,
+                                          "sec_lifetime": sec_lifetime}),
+                                 f"neon.check_for_signal.{signal_name}", 10) or Message('')
+    return stat.data.get("is_set")
+
+
+def _manager_wait_for_signal_create(signal_name: str, timeout: int = 30) -> bool:
+    """
+    Block until the specified signal is set or timeout is reached
+    :param signal_name: name of signal to check
+    :param timeout: max seconds to wait for signal to be created, Default is 30 seconds
+    :return: True if signal exists
+    """
+    timeout = 300 if timeout > 300 else timeout  # Cap wait at 5 minutes
+    bus_wait_time = timeout + 5  # Allow some padding for bus handler
+    stat = _BUS.wait_for_response(Message("neon.wait_for_signal_create",
+                                          {"signal_name": signal_name,
+                                          "timeout": timeout}),
+                                 f"neon.wait_for_signal_create.{signal_name}", bus_wait_time)
+    return stat.data.get("is_set")
+
+
+def _manager_wait_for_signal_clear(signal_name: str, timeout: int = 30) -> bool:
+    """
+    Block until the specified signal is cleared or timeout is reached
+    :param signal_name: name of signal to check
+    :param timeout: max seconds to wait for signal to be created, Default is 30 seconds
+    :return: True if signal exists
+    """
+    timeout = 300 if timeout > 300 else timeout  # Cap wait at 5 minutes
+    bus_wait_time = timeout + 5  # Allow some padding for bus handler
```

Comment:
I think better handle such things with config

Reply:
Good note, addressed in #241